### Day 15 of programming

### Python Tutorial: Math Functions
Introduction to Math in Python
Python provides a rich set of mathematical functions, both built-in and in the math module, which allows you to perform various mathematical operations like trigonometry, logarithms, exponents, and more.

You can also perform basic arithmetic operations (addition, subtraction, multiplication, division) without importing any module, as these operators are built into the Python language.

#### Step 1: Basic Arithmetic Operations

Python allows you to perform basic arithmetic operations using the following operators:

+: Addition

-: Subtraction

*: Multiplication

/: Division (floating-point result)

//: Integer (floor) division

%: Modulo (remainder)

**: Exponentiation (raising to a power)

#### Example: Basic Arithmetic Operations

In [1]:
# Addition
print(5 + 3)  # Output: 8

# Subtraction
print(10 - 4)  # Output: 6

# Multiplication
print(7 * 2)  # Output: 14

# Division (floating-point result)
print(15 / 4)  # Output: 3.75

# Integer (floor) division
print(15 // 4)  # Output: 3

# Modulo (remainder)
print(15 % 4)  # Output: 3

# Exponentiation (power)
print(2 ** 3)  # Output: 8


8
6
14
3.75
3
3
8


### Step 2: Advanced Math Functions with math Module
To use more advanced mathematical functions like square roots, trigonometric functions, logarithms, and constants, you need to import the math module.

Example: Importing the math Module

In [2]:
import math

# Find the square root of a number
print(math.sqrt(16))  # Output: 4.0

# Find the sine of an angle (in radians)
print(math.sin(math.pi / 2))  # Output: 1.0

# Find the logarithm (base 10) of a number
print(math.log10(1000))  # Output: 3.0

# Find the value of π (Pi)
print(math.pi)  # Output: 3.141592653589793

# Find the value of e (Euler's number)
print(math.e)  # Output: 2.718281828459045


4.0
1.0
3.0
3.141592653589793
2.718281828459045


### Step 3: Commonly Used math Functions
Here are some of the most commonly used functions from the math module:

#### 1. Square Root (sqrt())
The sqrt(x) function returns the square root of x.

In [3]:
import math
print(math.sqrt(25))  # Output: 5.0


5.0


#### 2. Power (pow())
The pow(x, y) function returns x raised to the power of y.

In [4]:
print(math.pow(2, 3))  # Output: 8.0


8.0


#### 3. Factorial (factorial())
The factorial(n) function returns the factorial of n, which is the product of all positive integers less than or equal to n.

In [5]:
print(math.factorial(5))  # Output: 120 (5 * 4 * 3 * 2 * 1)


120


#### 4.  Constants
math.pi: The value of π (approximately 3.14159).

math.e: The value of Euler’s number e (approximately 2.718).

In [6]:
print(math.pi)  # Output: 3.141592653589793
print(math.e)   # Output: 2.718281828459045


3.141592653589793
2.718281828459045


### Step 4: Rounding and Precision Functions
Python provides several functions to control the precision of numerical calculations:

round(x, n): Rounds x to n decimal places.

math.floor(x): Returns the largest integer less than or equal to x.
    
math.ceil(x): Returns the smallest integer greater than or equal to x.

In [7]:
import math

# Round 5.6789 to 2 decimal places
print(round(5.6789, 2))  # Output: 5.68

# Floor (round down)
print(math.floor(3.9))  # Output: 3

# Ceil (round up)
print(math.ceil(3.1))  # Output: 4


5.68
3
4


### Practice Question

1. Write a Python program that asks the user to input a number and prints the factorial of that number using math.factorial().
2. Write a Python function that simulates rolling a pair of dice. Use math.floor() and random.random() to generate two random numbers between 1 and 6, and print their sum.
3. Modify the trigonometric examples so that they work with degrees instead of radians. Write a program that takes an angle in degrees, converts it to radians, and then finds the sine, cosine, and tangent.



### Summary

Basic Arithmetic: Python supports basic arithmetic operators like +, -, *, /, **, etc.

math Module: The math module provides functions for more advanced mathematical operations like trigonometry, logarithms, factorials, and constants like pi and e.

Rounding and Precision: Functions like round(), math.floor(), and math.ceil() allow for precise control over numerical rounding.

By practicing these math operations, you'll be equipped to handle a variety of mathematical problems in Python.